src/main/scala/is/hail/expr/ir/functions/MathFunctions.scala

```diff
@@ -43,6 +44,17 @@ object MathFunctions extends RegistryFunctions {
 
   def floorDiv(x: Double, y: Double): Double = math.floor(x / y)
 
+  def approxEqual(x: Double, y: Double, tolerance: Double, absolute: Boolean): Boolean = {
+    x == y || {
+      (if (absolute)
+        math.abs(x - y) <= tolerance
+      else
+        D_==(x, y, tolerance)) ||
+        (x.isNaN && y.isNaN) ||
+        (x.isInfinite && y.isInfinite && ((x > 0 && y > 0) || (x < 0 && y < 0)))
```

Comment:
Sorry, didn't notice this the first time.

In the IEEE standard, NaN == NaN is false. I just checked, and the same is true in Python and Scala. So it would seem odd to have NaN approximately equal to itself.

On the other hand, the `x == y` check should handle infinities just fine.

Could you remove the last two lines from the else clause, and add a couple of test cases to cover NaN and infinities?

Reply:
I removed the last two lines for this user-exposed function. However, for the `valuesSimilar` comparison for TFloat32 and TFloat64, I kept the NaN comparison check because those methods are used in `_same` and to me it makes sense that we'd want to have True if both datasets have NaN at the same place.